src/actions.js

```diff
@@ -45,14 +43,26 @@ export const fetchItems = (url, dispatch) => {
           reject(new Error('Request timed out'));
         }, FETCH_TIMEOUT);
 
-        fetch(url, {
+        window.fetch(url, {
           credentials: 'include',
           headers: {
             Accept: 'application/json'
           }
         })
-          .json
-          .then(resolve, reject);
+          .then(response => {
+            return response.json()
+              .then(json => {
+                if (response.status > 399) {
+                  const err = new Error(json.message);
```

Comment:
We might want to add a default message here just in case the json doesn't have a `message` property.